stdlib/args.ysh

```diff
@@ -3,50 +3,50 @@
 # Usage:
 #   source --builtin args.sh
 #
-# Args :spec {
-#   flag -v --verbose "Verbosely"  # default is Bool, false
+# parser (&spec) {
+#   flag -v --verbose (help="Verbosely")  # default is Bool, false
 #
-#   flag -P --max-procs '''
+#   flag -P --max-procs ('int', default=-1, doc='''
 #     Run at most P processes at a time
-#     ''' (Int, default=-1)
+#     ''')
 #
-#   flag -i --invert '''
+#   flag -i --invert ('bool', default=true, doc='''
 #     Long multiline
 #     Description
-#     ''' (Bool, default = true)
+#     ''')
 #
-#   arg src 'Source'
-#   arg src 'Dest'
-#   arg times 'Foo' (Int)
+#   arg src (help='Source')
+#   arg dest (help='Dest')
+#   arg times (help='Foo')
+#
+#   rest files
 # }
 #
-# var opt, i = parseArgs(spec, ARGV)
+# var args = parseArgs(spec, ARGV)
```

Comment:
Having this code example here makes me wonder if we need docstrings in YSH, probably with `'''` like Python.  Since we already have that syntax.

It also reminds me that at some point we will move the tests from spec tests to an actual YSH test framework!    In pure YSH code, so people can read the tests for examples.

That will be exciting :)